server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -851,18 +849,24 @@ public void write(XContentBuilder b) throws IOException {
 
             if (ignoredValues != null && ignoredValues.isEmpty() == false) {
                 // Use an ordered map between field names and writer functions, to order writing by field name.
-                Map<String, CheckedConsumer<XContentBuilder, IOException>> orderedFields = new TreeMap<>();
+                Map<String, FieldWriter> orderedFields = new TreeMap<>();
                 for (IgnoredSourceFieldMapper.NameValue value : ignoredValues) {
-                    orderedFields.put(value.name(), value::write);
+                    var existing = orderedFields.get(value.name());
+                    if (existing == null) {
+                        orderedFields.put(value.name(), new FieldWriter.IgnoredSource(value));
+                    } else if (existing instanceof FieldWriter.IgnoredSourceWriter isw) {
+                        orderedFields.put(value.name(), isw.mergeWith(value));
+                    }
                 }
                 for (SourceLoader.SyntheticFieldLoader field : fields) {
                     if (field.hasValue()) {
                         // Skip if the field source is stored separately, to avoid double-printing.
```

Comment:
Iirc the use case for this is when `store_array_source` is set. We can check for that and merge otherwise.

Reply:
But we don't know which mapper produced this field loader at this point?

Reply:
In the current logic, the param would be set in the container `ObjectMapper` running this. So this could be:

```
if (field.hasValue()) {
  if (storeArraySource()) {
    // Skip if the field source is stored separately, to avoid double-printing.
    orderedFields.computeIfAbsent(field.fieldName(), k -> new FieldWriter.FieldLoader(field));
  } else {
   // Merge with existing values.. 
  } 
}
```

Would something like this work? We can look at the mixed case separately..

Reply:
As we discussed today i think we can defer handling mixed case for now.